### PR TITLE
Update fold_old_batch_norms.cc to accommodate 'NCHW' format.

### PR DESCRIPTION
**Pull request by joeyearsley**
Fixes the problem of using fused batch normalization and this transform, only shows up when using 'NCHW' as the default is 'NHWC'.

### CODE REVIEW

**Comment by joeyearsley** [comment]
I signed it!

File: tensorflow/tools/graph_transforms/fold_old_batch_norms.cc [hunk 1]
```diff
@@ -159,6 +159,7 @@ Status FuseScaleOffsetToConvWeights(const std::vector<float>& scale_values,
   NodeDef bias_add_node;
   bias_add_node.set_op("BiasAdd");
   bias_add_node.set_name(conv_output_name);
+  CopyNodeAttr(conv_node, "data_format", "data_format", &bias_add_node);
```
Comment (mingxingtan) [inline tensorflow/tools/graph_transforms/fold_old_batch_norms.cc]: This would fail the whole program if the conv_node doesn't have "data_format" attribute. You probably need to first verify it has the attribute "data_format" before calling CopyNodeAttr?

**Review by mingxingtan — CHANGES_REQUESTED** [review]

File: tensorflow/tools/graph_transforms/fold_old_batch_norms.cc (hunk 1 above)
Comment (joeyearsley) [reply tensorflow/tools/graph_transforms/fold_old_batch_norms.cc]: I don't see any checks like this in any other part of the code, please could you direct me to some.

Also, since it is now part of the api in `tf.nn.conv2d` does it not get this attribute regardless?  - I'm not sure but I thought that it would fallback to the default.

**Comment by joeyearsley** [comment]
@mingxingtan Added the check as requested.

**Comment by martinwicke** [comment]
@mingxingtan can you take another look?

File: tensorflow/tools/graph_transforms/fold_old_batch_norms.cc [hunk 2]
```diff
@@ -159,6 +159,9 @@ Status FuseScaleOffsetToConvWeights(const std::vector<float>& scale_values,
   NodeDef bias_add_node;
   bias_add_node.set_op("BiasAdd");
   bias_add_node.set_name(conv_output_name);
+  if (HasAttr(conv_node, "data_format")) {
```
Comment (mingxingtan) [inline tensorflow/tools/graph_transforms/fold_old_batch_norms.cc]: I don't see where is HasAttr defined. Maybe you can directly use something like:

```
if (!conv_node.attr().count("data_format")) {
  CopyNodeAttr(conv_node, "data_format", "data_format", &bias_add_node);
}
```

File: tensorflow/tools/graph_transforms/fold_old_batch_norms.cc (hunk 1 above)
Comment (mingxingtan) [reply tensorflow/tools/graph_transforms/fold_old_batch_norms.cc]: This is just an extra safety check for legacy tf graphs. You are right, If the original graph is created with tf.nn.conv2d, then this check is not needed; but I am not whether there are any legacy tf graphs that do not contain this attr.

File: tensorflow/tools/graph_transforms/fold_old_batch_norms.cc (hunk 2 above)
Comment (joeyearsley) [reply tensorflow/tools/graph_transforms/fold_old_batch_norms.cc]: Updated as requested

**Review by mingxingtan — APPROVED** [review]
